### DISCUSSION TITLE
How to Add TTS Functionality to RAGFlow?

### DISCUSSION THREAD

**Discussion by Chenjie669** [infiniflow/ragflow#2489] [Q&A]
Hi everyone,

I’m currently exploring the RAGFlow framework and would like to add Text-to-Speech (TTS) functionality to enhance the system by converting generated answers into speech.

After reviewing the architecture, I believe TTS should be integrated after the Answer is generated by the LLMs, but I’m unsure exactly where in the codebase this would be best implemented.

Could anyone guide me on which file or module I should edit to incorporate the TTS functionality?  Any suggestions on how to properly integrate a TTS library (e.g., Google TTS, Amazon Polly, or any open-source alternative) would be greatly appreciated!

Thanks in advance for your help!

**Comment by KevinHuSh** [comment]
Checkout this PR: https://github.com/infiniflow/ragflow/pull/2311/files
#2493
